src/plugins/dashboard/common/saved_dashboard_references.ts

```diff
@@ -17,18 +17,42 @@
  * under the License.
  */
 
-import { SavedObjectAttributes, SavedObjectReference } from 'kibana/public';
-import { SavedObjectDashboard } from './saved_dashboard';
+import { SavedObjectAttributes, SavedObjectReference } from '../../../core/types';
+import {
+  extractPanelsReferences,
+  injectPanelsReferences,
+} from './embeddable/embeddable_references';
+import { SavedDashboardPanel730ToLatest } from './types';
+import { EmbeddablePersistableStateService } from '../../embeddable/common/types';
 
-export function extractReferences({
-  attributes,
-  references = [],
-}: {
+export interface ExtractDeps {
+  embeddablePersistableStateService: EmbeddablePersistableStateService;
+}
+
+export interface SavedObjectAttributesAndReferences {
   attributes: SavedObjectAttributes;
   references: SavedObjectReference[];
-}) {
+}
+
+export function extractReferences(
+  { attributes, references = [] }: SavedObjectAttributesAndReferences,
+  deps: ExtractDeps
+): SavedObjectAttributesAndReferences {
   const panelReferences: SavedObjectReference[] = [];
-  const panels: Array<Record<string, string>> = JSON.parse(String(attributes.panelsJSON));
+  let panels: Array<Record<string, string>> = JSON.parse(String(attributes.panelsJSON));
+
+  const extractedReferencesResult = extractPanelsReferences(
+    (panels as unknown) as SavedDashboardPanel730ToLatest[],
+    deps
+  );
+
+  panels = (extractedReferencesResult.map((res) => res.panel) as unknown) as Array<
+    Record<string, string>
+  >;
+  extractedReferencesResult.forEach((res) => {
+    panelReferences.push(...res.references);
+  });
+
   panels.forEach((panel, i) => {
```

Comment:
can we create an issue to move this inside embeddable reference extraction (rather than having it here on dashboard) ?

Reply:
I added a comment and created a placeholder issue: https://github.com/elastic/kibana/pull/82602/commits/9cd13a8c75de98fd0175d07313b4ac58c5d9f8bd

I am not 100% sure though that it is possible without breaking changes and if this is required. This extract/inject now operates on dashboard panel specific data model and not on embeddable model.